tests/robottelo/test_vm.py

```diff
@@ -126,3 +127,48 @@ def test_destroy(self, ssh_command):
         ]
 
         self.assertListEqual(ssh_command.call_args_list, ssh_command_args_list)
+
+    @patch('time.sleep')
+    @patch('robottelo.ssh.command', side_effect=[
+        ssh.SSHCommandResult(),
+        ssh.SSHCommandResult(stdout=[
+            '{"return":[{"name":"lo","ip-addresses":[{"ip-address-type":"ipv4",'
+            '"ip-address":"127.0.0.1","prefix":8}],"hardware-address":"00:00:00:00:00:00"}'
+            ',{"name":"ens3","ip-addresses":[{"ip-address-type":"ipv4",'
+            '"ip-address":"10.8.30.135","prefix":19}],"hardware-address":"52:54:00:f7:bb:a8"}]}'
+        ]),
+        ssh.SSHCommandResult()
+    ])
+    def test_qemu_ga_gets_ip(self, ssh_command, sleep):
+        """Verify that the IP is correctly parsed from the qemu-guest-agent output"""
+        self.configure_provisioning_server()
+        vm = VirtualMachine()
+        vm.create()
+        self.assertEqual(vm.ip_addr, '10.8.30.135')
+
+    @patch('time.sleep')
+    @patch('robottelo.ssh.command', side_effect=[
+        ssh.SSHCommandResult(),
+        ssh.SSHCommandResult(
+            return_code=1,
+            stderr='error: Guest agent is not responding: '
+                   'QEMU guest agent is not connected'
+        ),
+        ssh.SSHCommandResult(
+            stdout=[
+                '{"return":[{"name":"lo","ip-addresses":'
+                '[{"ip-address-type":"ipv4","ip-address":"127.0.0.1","prefix":8}],'
+                '"hardware-address":"00:00:00:00:00:00"},{"name":"ens3","ip-addresses":'
+                '[{"ip-address-type":"ipv6","ip-address":"2620:52:0:800:5054:ff:fe31:d7d3",'
+                '"prefix":64},{"ip-address-type":"ipv6","ip-address":"fe80::5054:ff:fe31:d7d3"'
+                ',"prefix":64}],"hardware-address":"52:54:00:31:d7:d3"}]}']),
+        ssh.SSHCommandResult(stdout=[
+            '{"return":[{"name":"lo","ip-addresses":[{"ip-address-type":"ipv4",'
+            '"ip-address":"127.0.0.1","prefix":8}],"hardware-address":"00:00:00:00:00:00"}'
+            ',{"name":"ens3","ip-addresses":[{"ip-address-type":"ipv4",'
+            '"ip-address":"10.8.30.135","prefix":19}],"hardware-address":"52:54:00:f7:bb:a8"}]}'
+        ]),
+        ssh.SSHCommandResult()
+    ])
+    def test_qemu_ga_waits_for_ip(self, ssh_command, sleep):
+        """Verify that the qemu-ga waits for the IPv4 to be obtained"""
```

Comment:
Is there something that we could do to make this test more explicit? Because right now it looks like function that does nothing, because all the logic is moved into decorators and fixtures.

Reply:
Not sure what you'd like to see here. 
The purpose of the test is to assert that `vm.create()` properly acquires the ip address from the ssh command output, while the mocking simulates the possible qemu-ga outputs.

Reply:
we miss test body here?
```
        self.configure_provisioning_server()
        vm = VirtualMachine()
        vm.create()
        self.assertEqual(vm.ip_addr, '10.8.30.135')
```